scala/scala.bzl

```diff
@@ -747,23 +749,31 @@ def _scala_test_impl(ctx):
     )
     return out
 
-def _gen_test_suite_flags_based_on_prefixes_and_suffixes(ctx, archive):
+def _gen_test_suite_flags_based_on_prefixes_and_suffixes(ctx, scala_archive, maybe_java_archive):
+    archives = _get_archives_short_path(scala_archive, maybe_java_archive)
     return struct(testSuiteFlag = "-Dbazel.test_suite=io.bazel.rulesscala.test_discovery.DiscoveredTestSuite",
-    archiveFlag = "-Dbazel.discover.classes.archive.file.path=%s" % archive.short_path,
+    archiveFlag = "-Dbazel.discover.classes.archives.file.paths=%s" % archives,
     prefixesFlag = "-Dbazel.discover.classes.prefixes=%s" % ",".join(ctx.attr.prefixes),
     suffixesFlag = "-Dbazel.discover.classes.suffixes=%s" % ",".join(ctx.attr.suffixes),
     printFlag = "-Dbazel.discover.classes.print.discovered=%s" % ctx.attr.print_discovered_classes)
 
+def _get_archives_short_path(scala_archive, maybe_java_archive):
+  java_archive_short_path = ""
+  if maybe_java_archive:
```

Comment:
Maybe instead of this if, you can pass this method a list of archives, and some how archives that are None will not get into the list...

Reply:
I have a feeling it's indirection and not abstraction. There are just two instances and if I understand correctly you're suggestion is to still have the if just for a "generic" item, right?

Reply:
ended up doing it for a different reason, it allowed me to pass a more generic out argument from the _common_binary_common